spec/tasks/scheduler_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "rails_helper"
+require "rake"
+
+describe "scheduler.rake rake tasks" do
+  let(:task) { Rake::Task[task_name] }
+  let(:task_name) { "my:rake:task" }
+
+  after(:each) do
+    task.reenable
+  end
+
+  describe "monthly_price_summary" do
+    let(:task_name) { "monthly_price_summary" }
+
+    it "sends a monthly price summary" do
+      expect(HappyHood::Slack::Client).to receive(:send_monthly_price_summary)
+
+      task.invoke
+    end
+
+    context "with caching enabled" do
+      before do
+        file_store_cache = ActiveSupport::Cache.lookup_store(:file_store, "tmp/test#{ENV["TEST_ENV_NUMBER"]}/cache")
+        allow(Rails).to receive(:cache).and_return(file_store_cache)
+        Rails.cache.clear
+      end
```

Comment:
A little low-level explanation:

Because [`config/environments/config.rb#cache_store = :null_store`, very little/no caching is actually done](https://guides.rubyonrails.org/caching_with_rails.html#activesupport-cache-nullstore).

We opt for a `file_store` approach. If the suite is parallelized, each process will have access to the OS's file system to look up the cache. If this suite is run in parallel for whatever reason, a memory store wouldn't be enough because mem store operates on a per-process basis.

If we find ourselves needing this functionality even more, we can add some helpers to our rspec config to allow us to do something more fancy (like test case metadata to signify when to use the file cache store).